docs/extensiondev.rst

```diff
@@ -542,3 +542,227 @@ your HTTP requests::
 
 For further details, see Requests' docs on `session objects
 <http://www.python-requests.org/en/latest/user/advanced/#session-objects>`__.
+
+Testing extensions
+==================
+
+Creating test cases for your extensions makes them much simpler to maintain
+over the long term. It can also make it easier for you to review and accept
+pull requests from other contributors knowing that they will not break the
+extension in some unanticipated way.
+
+Before getting started, it is important to familiarize yourself with the
+Python `mock library <https://docs.python.org/dev/library/unittest.mock.html>`_.
+When it comes to running tests, Mopidy typically makes use of testing tools
+like `tox <https://tox.readthedocs.org/en/latest/>`_ and
+`pytest <http://pytest.org/latest/>`_.
+
+Testing approach
+----------------
+
+To a large extent the testing approach to follow depends on how your extension
+is structured, which parts of Mopidy it interacts with, and if it uses any 3rd
+party APIs or makes any HTTP requests to the outside world.
+
+The sections that follow contain code extracts that highlight some of the
+key areas that should be tested. For more exhaustive examples, you may want to
+take a look at the test cases that ship with Mopidy itself which covers
+everything from instantiating various controllers, reading configuration files,
+and simulating events that your extension can listen to.
+
+In general your tests should cover the extension definition, the relevant
+Mopidy controllers, and the Pykka backend and / or frontend actors that form
+part of the extension.
+
+Testing the extension definition
+--------------------------------
+Test cases for checking the definition of the extension should ensure that:
+- the extension provides a ``ext.conf`` configuration file containing the
+  relevant parameters with their default values,
+- that the config schema is fully defined, and
+- that the extension's actor(s) are added to the Mopidy registry on setup.
+
+An example of what these tests could look like is provided below::
+
+    def test_get_default_config(self):
+        ext = Extension()
+        config = ext.get_default_config()
+
+        self.assertIn('[my_extension]', config)
```

Comment:
We should use pytest's assert syntax instead, e.g. `assert '[my_extension]' in config`.